icu4c/source/i18n/unicode/numberformatter.h

```diff
@@ -2410,14 +2410,14 @@ class U_I18N_API LocalizedNumberFormatter
 class U_I18N_API FormattedNumber : public UMemory, public FormattedValue {
   public:
 
-#ifndef U_HIDE_DRAFT_API
+    // Default constructor cannot have #ifndef U_HIDE_DRAFT_API
```

Comment:
Because C++ will auto-generate one if it is conditionalized out. This is specifically called out in the [C coding guidelines](http://userguide.icu-project.org/dev/codingguidelines#TOC-C-C-Hiding-Un--stable-APIs).